### ISSUE TITLE
Two instances of squel will not handle subqueries correctly

### ISSUE THREAD

**Issue by rooftopsparrow** [hiddentao/squel#229]
I wrote a little test script to demonstrate the issue ( bash environment assumed ).

``` bash
mkdir test
cd test
mkdir node_modules
cd node_modules
mkdir test-lib
cd test-lib
cat <<EOF > index.js
var squel = require('squel')
module.exports = function testQuery() {
  return squel.select().from('things').where('a = ?', 1)
}
EOF
npm init -y
npm install squel -S
cd ../..
cat <<EOF > index.js
var testLib = require('test-lib')
var squel = require('squel')
var q = testLib()
q.where('b in ?', squel.select().field('b').from('otherthings'))
console.log('result => ', q.toString())
EOF
npm init -y
npm install squel -S
node index.js
# result =>  SELECT * FROM things WHERE (a = 1) AND (b in 'SELECT b FROM otherthings')
```

The above result shows the subquery is a string instead of an actual subquery. Once we eliminate the duplicate library it works:

``` bash
npm dedupe
node index.js
result =>  SELECT * FROM things WHERE (a = 1) AND (b in (SELECT b FROM otherthings))
```

What is concerning about this is when there are different versions of the same library, duplicate libraries are expected in the npm tree and the same issue will happen.

Let me know if I can assist in figuring this out or if what I've provided here is unclear. Thanks!

**Comment by hiddentao** [comment]
Squel uses `instanceof` checks within to know when something is a builder-derived object. I wonder if this is the cause of this.

**Comment by hiddentao** [comment]
This is a complex problem to solve. I tried adding string tags to the classes - as an alternative means of checking what an item is rather than using `instanceof`. But of course tags won't take inheritance into account. I'm going to leave this for now, unless more people state this as a problem. Typically I don't expect people to use multiple versions of squel in the same project.

**Comment by jbrawnjr** [comment]
I recently discovered a similar issue, which I believe is related.

```javascript
squel.useFlavour('postgres').insert().into('hi').fromQuery(['name'], squel.select().from('bye').field('name')).toString()
// > Uncaught Error: must be a builder instance(…)

squel.useFlavour('postgres').insert().into('hi').fromQuery(['name'], squel.useFlavour('postgres').select().from('bye').field('name')).toString()
// > Uncaught Error: must be a builder instance(…)

// but...
sq = squel.useFlavour('postgres')
sq.insert().into('hi').fromQuery(['name'], sq.select().from('bye').field('name')).toString()
// > "INSERT INTO hi (name) (SELECT name FROM bye)"
```

**Comment by hiddentao** [comment]
Fixed in 5.5.1
